app/models/frontend/TaskList.scala

```diff
@@ -30,19 +30,20 @@ object TaskList {
 
   val empty: TaskList = TaskList(None, Nil, None)
 
-  def fromJourneyAnswers(userJourneyAnswers: List[JourneyAnswers], businesses: List[Business]): TaskList = {
-    val groupedByBusinessId: Map[BusinessId, Seq[JourneyAnswers]] = userJourneyAnswers.groupBy(_.businessId)
-    val tradingDetailsStatus = groupedByBusinessId
-      .get(BusinessId.tradeDetailsId)
-      .flatMap(
-        _.toList.headOption
-          .map(a => JourneyNameAndStatus(a.journey, a.status)))
+  def fromJourneyAnswers(userJourneyAnswers: List[JourneyAnswers], businesses: List[Business], mtditid: Mtditid): TaskList = {
+    def getStatusWithoutBusinessId(journeyName: JourneyName): Option[JourneyNameAndStatus] = {
+      val groupedByJourney = userJourneyAnswers.groupBy(_.journey)
+      groupedByJourney
+        .get(journeyName)
+        .flatMap(_.filter(_.mtditid == mtditid).headOption.map(a => JourneyNameAndStatus(a.journey, a.status)))
```

Comment:
I don't think you will have any elements with different mtditid because the caller already called filterAllJourneys which include mtditid. Can you check if filter mtditid == mtditid part is redundant here because of

```
  private def filterAllJourneys(taxYear: TaxYear, mtditid: Mtditid): Bson = Filters.and(
    Filters.eq("mtditid", mtditid.value),
    Filters.eq("taxYear", taxYear.endYear)
  )
```

Reply:
You are right, have pushed the change